src/api/spec/controllers/webui/workflow_runs_controller_spec.rb

```diff
@@ -13,22 +13,74 @@
         get :index, params: { token_id: workflow_token.id }
       end
 
-      it { expect(assigns(:workflow_runs).count).to eq(1) }
+      it { expect(assigns(:workflow_runs)).to contain_exactly(workflow_run) }
     end
 
     context 'when action_filter is available' do
       it 'finds workflow runs when the action is available' do
-        get :index, params: { token_id: workflow_token.id, request_action: 'opened', generic_event_type: 'pull_request' }
+        get :index, params: { token_id: workflow_token.id, request_action: 'opened', pull_merge_request: 1 }
 
-        expect(assigns(:workflow_runs).count).to eq(1)
-        expect(assigns(:request_action)).to eq('opened')
+        expect(assigns(:workflow_runs)).to contain_exactly(workflow_run)
       end
 
       it 'does not find workflow runs when action is not available' do
-        get :index, params: { token_id: workflow_token.id, request_action: 'closed', generic_event_type: 'pull_request' }
+        # TODO: Check this pull_merge_requet parameter...
+        get :index, params: { token_id: workflow_token.id, request_action: 'closed', pull_merge_request: 1 }
 
-        expect(assigns(:workflow_runs).count).to eq(0)
-        expect(assigns(:request_action)).to eq('closed')
+        expect(assigns(:workflow_runs)).to be_empty
+      end
+    end
+
+    context 'when having no filters selected' do
+      let!(:pull_request_1_workflow_run) { create(:workflow_run, :succeeded) }
+      let(:token) { pull_request_1_workflow_run.token }
+      let!(:pull_request_2_workflow_run) { create(:workflow_run, :succeeded, event_source_name: '2', token: token) }
+      let!(:closed_pull_request_workflow_run) { create(:workflow_run, :pull_request_closed, token: token) }
+      let!(:running_push_workflow_run) { create(:workflow_run, :running, :push, token: token) }
+      let!(:succeeded_tag_push_workflow_run) { create(:workflow_run_gitlab, :failed, :tag_push, token: token) }
+
+      before do
+        login pull_request_1_workflow_run.token.executor
+      end
+
+      # rubocop:disable RSpec/ExampleLength
+      it 'shows all the workflow runs' do
+        get :index, params: { token_id: token.id, pr_mr: '', commit_sha: '' }
+
+        expect(assigns[:workflow_runs]).to contain_exactly(pull_request_1_workflow_run,
```

Comment:
Maybe create one more WorkflowRun that doesn't match?

Reply:
There is already a workflow run at the top of the spec that also applies here but does not match because it gets filtered out by the token.

Reply:
A bit implicit but okay good.